Quadriga/src/main/java/edu/asu/spring/quadriga/rest/SearchNetworkRestController.java

```diff
@@ -108,10 +113,19 @@
             return new ResponseEntity<String>(errorMsg, HttpStatus.NOT_FOUND);
         }
 
+        // Fetch ConceptPower entries related to the conceptId
+        ConceptpowerReply reply = conceptpowerConnector.getById(conceptId);
+        List<String> alternativeIdsForConcept = null;
+        if (reply != null && reply.getConceptEntry().size() > 0) {
+            alternativeIdsForConcept = reply.getConceptEntry().get(0).getAlternativeIdList();
+        }
+        if(alternativeIdsForConcept == null){
+            alternativeIdsForConcept = new ArrayList<String>();
+        }  
```

Comment:
Image it this way: if we were to switch frameworks (which has happened before) and not use Spring anymore, ideally (if we ignore that we've annotated all of our classes) we would just rewrite the controllers a bit to fit with another framework (preparing the data for the view). This would be really easy if all the controllers do is just preparing data for display, receiving data needed for executing logic and prepare the data for passing it to the service layer.

Not sure what you mean by  "trade-off between abstraction and dealing with dependencies". In the case right now, there isn't really a trade off. There are fewer dependencies to worry about in the controller and no abstraction is lost.